### PR TITLE
fast installation steps: define PYENV_PYTHON_MIRROR_URL variable.

### PR DESCRIPTION
**Pull request by Puins**
In China,source url download speed is slow, add third-party open source url.

Add PYENV_PYTHON_MIRROR_URL variable ,Improving download speed.

### CODE REVIEW

**Comment by hongyi-zhao** [comment]
Doesn't take effect, see following:

```
$ env PYENV_PYTHON_MIRROR_URL=https://cdn.npm.taobao.org/dist/python pyenv install -v 2.7.7
/tmp/python-build.20200326150107.13965 ~/.repo/github.com/pyenv/pyenv.git
Downloading Python-2.7.7.tar.xz...
-> https://www.python.org/ftp/python/2.7.7/Python-2.7.7.tar.xz
^Cerror: failed to download Python-2.7.7.tar.xz

```

**Comment by Mark24Code** [comment]
good job

File: plugins/python-build/bin/python-build [hunk 1]
```diff
@@ -480,7 +480,13 @@ reuse_existing_tarball() {
 }
 
 download_tarball() {
-  local package_url="$1"
+  #replaces python sources url
+  if [ -n "$PYENV_PYTHON_MIRROR_URL" ]; then
+      local package_url="$(echo "$1" | sed -r "s#(.*//www.python.org/ftp/python)*#$PYENV_PYTHON_MIRROR_URL#")"
```
Comment (joshfriend) [inline plugins/python-build/bin/python-build:485]: This only works for CPython versions.

pyenv can install numerous other types including Jython, PyPy, Stackless, miniconda, etc. and this solution will fail silently if you try to install one of those versions from a mirror

Perhaps a better solution would be one where the entire URL and base path of the python source/binary is replace

**Comment by joshfriend** [comment]
I would rather the original mirror mechanism be fixed as opposed to adding in an entirely new one

**Pull request closed without merging**